core/src/main/java/bisq/core/payment/PaymentAccountManager.java

```diff
@@ -0,0 +1,128 @@
+ * FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero General Public
+ * License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with Bisq. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package bisq.core.payment;
+
+import bisq.core.account.witness.AccountAgeWitnessService;
+import bisq.core.exceptions.NotFoundException;
+import bisq.core.exceptions.ValidationException;
+import bisq.core.locale.CryptoCurrency;
+import bisq.core.locale.CurrencyUtil;
+import bisq.core.locale.FiatCurrency;
+import bisq.core.locale.TradeCurrency;
+import bisq.core.offer.Offer;
+import bisq.core.offer.OpenOfferManager;
+import bisq.core.payment.validation.AltCoinAddressValidator;
+import bisq.core.trade.TradeManager;
+import bisq.core.user.Preferences;
+import bisq.core.user.User;
+import bisq.core.util.validation.InputValidator;
+
+import com.google.inject.Inject;
+
+import java.util.List;
+
+import static java.lang.String.format;
+
+public class PaymentAccountManager {
+
+    private final AccountAgeWitnessService accountAgeWitnessService;
+    private final OpenOfferManager openOfferManager;
+    private final Preferences preferences;
+    private final TradeManager tradeManager;
+    private final User user;
+    private AltCoinAddressValidator altCoinAddressValidator;
+
+    @Inject
+    public PaymentAccountManager(AccountAgeWitnessService accountAgeWitnessService,
+                                 AltCoinAddressValidator altCoinAddressValidator,
+                                 OpenOfferManager openOfferManager,
+                                 Preferences preferences,
+                                 TradeManager tradeManager,
+                                 User user) {
+        this.accountAgeWitnessService = accountAgeWitnessService;
+        this.altCoinAddressValidator = altCoinAddressValidator;
+        this.openOfferManager = openOfferManager;
+        this.preferences = preferences;
+        this.tradeManager = tradeManager;
+        this.user = user;
+    }
+
+    public PaymentAccount addPaymentAccount(PaymentAccount paymentAccount) {
+        if (paymentAccount instanceof CryptoCurrencyAccount) {
+            CryptoCurrencyAccount cryptoCurrencyAccount = (CryptoCurrencyAccount) paymentAccount;
+            TradeCurrency tradeCurrency = cryptoCurrencyAccount.getSingleTradeCurrency();
+            if (tradeCurrency == null) {
+                throw new ValidationException("CryptoCurrency account must have exactly one trade currency");
+            }
+            altCoinAddressValidator.setCurrencyCode(tradeCurrency.getCode());
+            InputValidator.ValidationResult validationResult = altCoinAddressValidator.validate(cryptoCurrencyAccount.getAddress());
+            if (!validationResult.isValid) {
+                throw new ValidationException(validationResult.errorMessage);
+            }
+        }
+//        TODO we should validate payment account here as well
```

Comment:
Is it out of scope of current pull request ?

Reply:
> Maybe removePaymentAccount method can return boolean directly ?

That would be consistent with classical `remove` methods which return a boolean as well about the success of the intended remove. In that case the client (e.g. UI) need to know why it was not possible. For that I recommend using custom methods.